cosmos/transaction.go

```diff
@@ -120,13 +120,16 @@ func (txn *Transaction) Get(partitionValue interface{}, id string, target Model)
 		// do nothing, cacheGet already unserialized to target
 	} else {
 		// post-get hook will be done by Collection.get()
-		err = txn.session.Collection.get(
+		response, err := txn.session.Collection.get(
```

Comment:
Hmm, I have one concern here. `Collection.get()` fires user-defined hook, and what if the cosmos client is used inside of that one to fetch another document?...

Ideally the session token would be populated after getting document, but before deserializing it...

Reply:
However, obviously our code right now doesn't depend on this. So wondering if we should fix this or not. Can't remember if the hook is able to access txn.session

Reply:
Are you referring to the post-get hook? Session token is set before the post-get hook is called. If the client is used within the post-get hook it will be used with the newest session. Which is what we want, right?

Reply:
hmm, you are right, I was confused by that comment, but I see now that post-get-hook is called further down